lib/devices/ios/ios-log.js

```diff
@@ -44,91 +40,112 @@ IosLog.prototype.startCapture = function (cb) {
   this.onIosLogStart = cb;
   // Select cmd for log capture
   if (this.udid) {
-    var spawnEnv = _.clone(process.env);
-    var limdDir = path.resolve(__dirname,
-                               "../../../build/deviceconsole");
-    spawnEnv.PATH = process.env.PATH + ":" + limdDir;
-    spawnEnv.DYLD_LIBRARY_PATH = limdDir + ":" + process.env.DYLD_LIBRARY_PATH;
-    logger.debug("Starting iOS device log capture via deviceconsole");
-    // deviceconsole retrieves many old device log lines that came before it was
-    // started so filter those out until we encounter new log lines.
     this.loggingModeOn = false;
-    this.proc = spawn("deviceconsole", ["-u", this.udid], {env: spawnEnv});
-    this.finishStartingLogCapture(cb);
-  } else {
-    var ver = parseInt(this.xcodeVersion.split(".")[0], 10);
-    if (ver >= 5) {
-      var logsPath;
-      if (ver >= 6) {
-        logger.debug("Starting iOS 8.* simulator log capture");
-        if (_.isUndefined(this.simUdid)) {
-          return cb(new Error("iOS8 log capture requires a sim udid"));
-        }
-        logsPath = path.resolve(process.env.HOME, "Library", "Logs",
-                                "CoreSimulator", this.simUdid);
+    var spawnEnv = _.clone(process.env);
+
+    logger.debug("Attempting iOS device log capture via libimobiledevice idevicesyslog");
+    which("idevicesyslog", function (err) {
+      if (!err) {
+        this.proc = spawn("idevicesyslog", {env: spawnEnv});
```

Comment:
this was already a bug in this file, but it's worth pointing out that I think `spawn` can throw synchronously, so it may be a good idea for us to wrap this and other instances in try/catch.